src/v/coproc/api.cc

```diff
@@ -58,10 +59,11 @@ ss::future<> api::start() {
     co_await _mt_frontend.start_single(std::ref(_rs.topics_frontend));
     co_await _pacemaker.start(_engine_addr, std::ref(_rs));
     co_await _pacemaker.invoke_on_all(&coproc::pacemaker::start);
-    _listener = std::make_unique<wasm::event_listener>();
 
+    _listener = std::make_unique<wasm::event_listener>(_as);
+    _dispatcher = std::make_unique<wasm::script_dispatcher>(_pacemaker, _as);
     _wasm_async_handler = std::make_unique<coproc::wasm::async_event_handler>(
-      _listener->get_abort_source(), std::ref(_pacemaker));
+      std::ref(*_dispatcher));
```

Comment:
I know `start` should only be called once, but I think it's worth asserting that a bunch of these pointers are null at the beginning of this function.